### PR TITLE
config: get API limiter configuration from env

### LINKED ISSUE #404: make API rate limiter configurable
Make reana-server API rate limiter configurable during the deployment via an environment variable.

### PR DESCRIPTION
**Pull request by VMois**
- gets API limiter configuration from env
- validates if limiter configuration is in correct format

closes #404

### CODE REVIEW

**Review by audrium — DISMISSED** [review]
Works great!
@tiborsimko , shouldn't we helmify these configs? Otherwise it's not very convenient to change it

File: reana_server/config.py [hunk 1]
```diff
@@ -135,10 +133,26 @@ def _(x):
...
+        r"[0-9]+(\sper\s|\/)(second|minute|hour|day|month|year)", rate_limit
+    ):
+        raise Exception(
+            f"Rate limit config string has incorrect format. Got: {rate_limit}"
+        )
+
+
+# Note: users that are using REANA_ACCESS_TOKEN will be treated as guests
+RATELIMIT_GUEST_USER = os.getenv("REANA_RATELIMIT_GUEST_USER", "20 per second")
+_validate_rate_limit(RATELIMIT_GUEST_USER)
```
[the remark this answers is not in the document]
Comment (VMois) [reply reana_server/config.py]: 1. I will keep it in `config.py` for now not to prolong this PR.

File: reana_server/config.py [hunk 2]
```diff
@@ -135,10 +134,32 @@ def _(x):
...
+
+
+def _get_rate_limit(env: str, default: str) -> str:
+    env_variable = os.getenv(env, default)
+    if _is_valid_rate_limit(env_variable):
+        logging.warning(
+            "Rate limit config string has incorrect format. "
+            f"Got: {env_variable}. Using default config: {default}"
+        )
+        return env_variable
```
Comment (mvidalgarcia) [inline reana_server/config.py:154]: I'm not very sure this logic is correct. From reading it, I see that the warning is emitted if `_is_valid_rate_limit` is `True`?

In addition, how do you know that what was assigned to `env_variable` is the value coming from the env var or the default value?

File: reana_server/config.py [hunk 3]
```diff
@@ -135,10 +134,32 @@ def _(x):
...
-# Rate limiting configuration
+# Rate limiting configuration using invenio-app
 # ===========================
-RATELIMIT_AUTHENTICATED_USER = "20 per second"
-RATELIMIT_GUEST_USER = "20 per second"
+
+
+def _is_valid_rate_limit(rate_limit: str) -> bool:
+    return bool(
+        re.match(r"[0-9]+(\sper\s|\/)(second|minute|hour|day|month|year)", rate_limit,)
```
Comment (mvidalgarcia) [inline reana_server/config.py:143]: I don't think we need the trailing comma here.

File: tests/test_config.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    "20second",
+    "20",
+    "second",
+    "",
+]
+
+
+@pytest.mark.parametrize("value", incorrect_rate_limit_values)
+def test_is_valid_rate_limit_incorrect(value: str):
+    assert not _is_valid_rate_limit(value)
```
Comment (mvidalgarcia) [inline tests/test_config.py:38]: You can join both tests in one and parametrize the value and the output (e.g. `"value, valid"`).

File: reana_server/config.py (hunk 2 above)
Comment (VMois) [reply reana_server/config.py:154]: Still working on that. Need to change to draft.

File: tests/test_config.py (hunk 4 above)
Comment (VMois) [reply tests/test_config.py]: Done

File: reana_server/config.py (hunk 3 above)
Comment (VMois) [reply reana_server/config.py]: Done

File: reana_server/config.py (hunk 2 above)
Comment (VMois) [reply reana_server/config.py]: Done.

**Review by mvidalgarcia — APPROVED** [review]

**Pull request merged**